peerassistant/health_checker.go

```
@@ -31,22 +35,14 @@ func init() {
 	}
 }
 
-func isHealthy(machineName string) poisonPillApis.HealthCheckResponse {
-	machine, err := client.Resource(machineRes).Namespace(machineNamespace).Get(context.TODO(), machineName, metav1.GetOptions{})
+func isHealthy(nodeName string) poisonPillApis.HealthCheckResponse {
+	_, err := client.Resource(pprRes).Namespace(pprNamespace).Get(context.TODO(), nodeName, metav1.GetOptions{})
 	if err != nil {
-		//todo examine the error, maybe it's 404 etc.
+		if apiErrors.IsNotFound(err) {
+			return poisonPillApis.Healthy
+		}
 		return poisonPillApis.ApiError
```

I suggest to wrap this error with the underlying one.

---

why?
the absence of the CR means the node is healthy

---

re-reading this I realized you pointed the api-error and not the healthy response, so now I understand your comment.

ATM I don't see a real need to propagate the error reason, and this could possibly increase the payload size + make the api a bit more complicated which is something I was trying to avoid for now.